registry.go

```diff
@@ -0,0 +1,68 @@
+package multihash
+
+import (
+	"crypto/md5"
+	"crypto/sha1"
+	"crypto/sha256"
+	"crypto/sha512"
+	"hash"
+)
+
+// registry is a simple map which maps a multihash indicator number
+// to a standard golang Hash interface.
+//
+// Multihash indicator numbers are reserved and described in
+// https://github.com/multiformats/multicodec/blob/master/table.csv .
+// The keys used in this map must match those reservations.
+//
+// Hashers which are available in the golang stdlib will be registered automatically.
+// Others can be added using the Register function.
+var registry = make(map[uint64]func() hash.Hash)
```

Comment:
Thank you for bringing up this whole range of questions.  (They are good.  So far, I made these diffs while trying to evade them!  But that does not represent a stance so much as it was just short-term diff-scope-minimization.)

This came up in enough places throughout this PR that I opened a whole new issue to talk through it: https://github.com/multiformats/go-multihash/issues/137

Reply:
... okay, my time & energy budget for recursing up through that concern tree has expired.

- I'm going to keep `uint64` as a map key (rather than any typedefs) because that's what this repo did before;
- I'm not going to add the https://github.com/multiformats/go-multicodec repo for symbols right now because it's not in this repo right now;
- I'm going to replace all the current use of bare constants with the named constants already in this package and leave them as they are.
- I'm going to make a concerted effort to do nothing to the `Names` and `Codes` maps in the rest of this PR because the things we'd probably want to do with them would almost certainly involve bringing in the above, which... E_SCOPE_LIMIT.

Additional boat rocking about constants can come some other day and some other PR.